backend/src/modules/tournament-tree/core/domain/TournamentTree.ts

```diff
@@ -0,0 +1,91 @@
+import { TournamentTeam } from './TournamentTeam';
+import { FightingTeamsGroup } from './FightingTeamsGroup';
+import { WinnerTree } from './WinnerTree';
+import { EntityIdGenerator } from '../../../../shared/core/application/EntityIdGenerator';
+import { CreateTournamentTree } from '../application/command/CreateTournamentTree';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+import { CurrentTimeProvider } from '../../../../shared/core/CurrentTimeProvider';
+import { TournamentTreeWasCreated } from './event/TournamentTreeWasCreated';
+
+export class TournamentTree {
+  readonly tournamentTeams: TournamentTeam[];
+  readonly tournamentTreeArray: FightingTeamsGroup[];
+  readonly tournamentId: string;
+
+  private constructor(props: { tournamentId: string; tournamentTreeArray: FightingTeamsGroup[]; tournamentTeams: TournamentTeam[] }) {
```

Comment:
torchę tutaj nie rozumiem tej zasady z prywatnymi konstruktorami i statycznymi metodami. Przecież to jest nasz główny obiekt domenowy to chyba normalnie będzie robić new TournamentTree? Czy to jest jednak dobre podejście?
@nowakprojects